slave_poller.go

```diff
@@ -157,6 +157,8 @@ func slavePoller(c *http.Client, conf *Config, frameworkRegistry *frameworkRegis
 			prometheus.Unregister(memRssGauge)
 
 			log.Errorf("Error retrieving stats from slave '%s' - Stopping goroutine", slave.Pid)
+
+			(*erroredSlaves)[slave.Pid] = struct{}{}
```

Comment:
Map are not save when accessed concurrently as described [here](https://blog.golang.org/go-maps-in-action#TOC_6.). Like in the article, I would suggest using a mutex. You can use the [frameworkRegistry struct](https://github.com/wndhydrnt/mesos-task-exporter/blob/master/framework_registry.go) as a template.